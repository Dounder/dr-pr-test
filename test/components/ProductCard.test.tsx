import * as React from 'react';
import renderer from 'react-test-renderer';

import ProductCard, { ProductTitle } from '../../src/components';
import { produc2 } from '../data/products';

const { act } = renderer;

describe('ProductTitle', () => {
  test('should show the component correctly', () => {
    const wrapper = renderer.create(
      <ProductCard product={produc2}>{() => <ProductTitle />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should increment counter', () => {
    const wrapper = renderer.create(
      <ProductCard product={produc2}>
        {({ count, increaseBy }) => (
          <>
            <h1>ProductCard</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}></button>
          </>
        )}
      </ProductCard>
    );

    let tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();

    act(() => (tree as any).children[2].props.onClick());

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');
  });
});
