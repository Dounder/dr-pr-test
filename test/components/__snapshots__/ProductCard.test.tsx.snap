// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should increment counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductTitle should show the component correctly 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - With Image
  </span>
</div>
`;
